Eternal Screaming (Def Updates)

Kuroitsune opened the pull request:
[ALL] Service Flags - Added 'Repair' to the list (9) for TES3.
[TES3] ALL\Header - Updated Record Flags format to function like other game definitions.
[TES3] ALL - Fixed various spelling errors.
[TES3] TES3\HEDR - Updated 'Version' float variable to match the Construction Set.
[TES3] ALCH\ALDT - Changed name to 'Data' to be consistent with other form data specific signatures.
[TES3] ALCH\ALDT - Changed 'Value' to 'Potion Value' to better reflect the Construction Set.
[TES3] BODY\BYDT - Changed 'Part' to 'Body Part' to differentiate from Part Type.
[TES3] BOOK\BKDT - Changed name to 'Data' to be consistent with other form data specific signatures.
[TES3] CELL\NAME - Changed name to 'Location' since CELLs don't really have an Editor ID.
[TES3] CLAS\CLDT - Restructured the definition to be more readable.
[TES3] INFO\QSTN - Changed name to 'Quest Named' to be past tensed like the other two signatures in the struct.
[TES3] LAND\VNML,VHGT,WNAM,VCLR,VTEX - Restructured the signatures to better match other game definitions.
[TES3] LAND\WNAM - Changed name to 'World Map Colors' because that makes more sense than World Map Painting.
[TES3] LIGH\LHDT - Corrected 'Radius' from a float to a signed 32 bit integer.
[TES3] MGEF\MEDT - Replaced the 'Color' struct with a custom wbByteColors.
[TES3] MISC\MCDT - Changed 'Is Key' to 'Is Used As Key.'
[TES3] PGRD\NAME - Changed to 'Location' to match CELL's NAME.
[TES3] PGRD\PGRP - Replaced the 'Position' struct with a custom wbVec3Pos.
[TES3] REFR\CNDT - Changed name to 'New Cell Owner' since MVRFs are stored in the original cell, not the cell the reference was moved to.
[TES3] REFR\DATA - Changed name to 'Data' for simplicity.
[TES3] SCPT\SetToStr - Added wbScriptToStr and updated it to work with TES3 formatted scripts.
[TES3] SKIL\SKDT - Updated 'Actions' floats to match decimal points used in the Construction Set.
[TES3] SPEL - Changed form type name to Spell to make it simple and align with other game definitions.
[TES3] WEAP\WPDT - Updated 'Speed' and 'Reach' floats to match decimal points used in the Construction Set.
[TES4] CSTY\CSTD,CSAD - Grouped applicable definitions together to make it more readable.
[TES4] EFSH\DATA - Grouped applicable definitions together to make it more readable.
[TES4] HAIR\MODL - Changed name from Skeleton to Model.
[TES4] MGEF\EDID - Forwarded wbStringMgefCode to account for Oblivion Magic Extender Magic Effects EDIDs being four bytes long with terminating zeros in them.

Kuroitsune committed:
Eternal Screaming (Def Updates)

[ALL] Service Flags - Added 'Repair' to the list (9) for TES3.
[TES3] ALL\Header - Updated Record Flags format to function like other game definitions.
[TES3] ALL - Fixed various spelling errors.
[TES3] TES3\HEDR - Updated 'Version' float variable to match the Construction Set.
[TES3] ALCH\ALDT - Changed name to 'Data' to be consistent with other form data specific signatures.
[TES3] ALCH\ALDT - Changed 'Value' to 'Potion Value' to better reflect the Construction Set.
[TES3] BODY\BYDT - Changed 'Part' to 'Body Part' to differentiate from Part Type.
[TES3] BOOK\BKDT - Changed name to 'Data' to be consistent with other form data specific signatures.
[TES3] CELL\NAME - Changed name to 'Location' since CELLs don't really have an Editor ID.
[TES3] CLAS\CLDT - Restructured the definition to be more readable.
[TES3] INFO\QSTN - Changed name to 'Quest Named' to be past petensed like the other two signatures in the struct.
[TES3] LAND\VNML,VHGT,WNAM,VCLR,VTEX - Restructured the signatures to better match other game definitions.
[TES3] LAND\WNAM - Changed name to 'World Map Colors' because that makes more sense than World Map Painting.
[TES3] LIGH\LHDT - Corrected 'Radius' from a float to a signed 32 bit integer.
[TES3] MGEF\MEDT - Replaced the 'Color' struct with a custom wbByteColors.
[TES3] MISC\MCDT - Changed 'Is Key' to 'Is Used As Key.'
[TES3] PGRD\NAME - Changed to 'Location' to match CELL's NAME.
[TES3] PGRD\PGRP - Replaced the 'Position' struct with a custom wbVec3Pos.
[TES3] REFR\CNDT - Changed name to 'New Cell Owner' since MVRFs are stored in the original cell, not the cell the reference was moved to.
[TES3] REFR\DATA - Changed name to 'Data' for simplicity.
[TES3] SCPT\SetToStr - Added wbScriptToStr and updated it to work with TES3 formatted scripts.
[TES3] SKIL\SKDT - Updated 'Actions' floats to match decimal points used in the Construction Set.
[TES3] SPEL - Changed form type name to Spell to make it simple and align with other game definitions.
[TES3] WEAP\WPDT - Updated 'Speed' and 'Reach' floats to match decimal points used in the Construction Set.
[TES4] ALL - Fixed various spelling errors.
[TES4] CSTY\CSTD,CSAD - Grouped applicable definitions together to make it more readable.
[TES4] EFSH\DATA - Grouped applicable definitions together to make it more readable.
[TES4] HAIR\MODL - Changed name from Skeleton to Model.
[TES4] MGEF\EDID - Forwarded wbStringMgefCode to account for Oblivion Magic Extender Magic Effects EDIDs being four bytes long with terminating zeros in them.

Kuroitsune committed:
Update xeInit.pas

zilav commented on Core/wbDefinitionsCommon.pas:
```diff
@@ -2860,12 +2862,20 @@ procedure wbScriptToStr(var aValue: string; aBasePtr: Pointer; aEndPtr: Pointer;
   if not wbTrySetContainer(aElement, aType, CER) then
     Exit;
 
+  var eSCDT := CER.ElementBySignature[SCDT];
```
`var eSCDA := CER.ElementBySignature[IfThen(wbGameMode <> gmTES3, SCDA, SCDT)];`

Other code changes aren't needed here. Just need to make overloaded IfThen for TwbSignature if not yet.

Kuroitsune replied:
I always forget the power of IfThen. I'll modify it when I wake up.

Kuroitsune committed:
The Power of IfThen

Kuroitsune committed:
Quests, Headers, & Flags - oh my.

[TES4] TOFT,TLOD - Removed commented references. Neither are recognized in the Construction Set. Comment1*
[TES4] TES4\OFST - Decoded as Form Load Order of form types. The same as the one in the exe. Ignored and collapsed by default.
[TES4] TES4\DELE - Decoded as Time Stamp. Ignored. Comment2*
[TES4] CREA\Header - Flag Unknown 19 is 'Starts Dead.'
[TES4] NPC_\Header - Flag Unknown 19 is 'Starts Dead.'
[TES4,FO3] DIAL\QSTI - Changed name to 'Associated Quests.' - wbQSTI added to Common Defs.
[TES4,FO3] DIAL\QSTR - Decoded as 'Removed Quests'. Same format as QSTI. - wbQSTR added to Common Defs. Comment3&4*

Comment1*
{LOCK, REPA, SNDG, TLOD, and TOFT appear in the load order list in the exe or the OFST in the official master file, but are not recognized by the Construction Set. All but TOFT will crash the vanilla CS and with the extender it simply throws an error about unrecognized forms for all five. Interesting fact: AMMO is referenced in the Morrowind.exe but is not recognized in the CK.}

Comment2*
{Only the Oblivion.esm has a value for DELE, all esm saved in the CK with or without the extender always save 8 bytes - all 00. If the value from the Oblivion.esm is tossed into HxD it comes up as 10/13/2004 17:05:47 in the Windows FILETIME. That was in the middle of Oblivion's production, most likely left overs from an earlier time when the time stamp replaced the Master Size (DATA) from Morrowind as a form of version control. Or it's just a concidence with random byte data. :peeposhrug:}

Comment3*
{If a quest from the Associated Quests no longer has any INFOs attached to the DIAL, either because the INFOs were moved to another topic or their assigned quest(QSTI) was changed, the Construction Set/GECK adds a QSTR for the quest no longer associated with the DIAL. Doesn't seem to have an affect in the game, both QSTI and QSTR are possibly just filters for quest dialogue in the editors.}

Comment4*
{Need to check if New Vegas works the same way as TES4 or FO3, or QSTR also contains Shared Info data like QSTI.}

Kuroitsune committed:
Quickie to restore script source if not

Kuroitsune committed:
Mostly TES3 again

TwbDivDef - Added ability to limit number the of digits.
[TES3] - Simplified several more definitions. [ENAM, Effects, ITEX, DESC, Items, Spells]
[TES3] TES3\FLTVR - Removed the Union to better match the other games. "It's all floats? Always has been."
[TES3] Flags - Added dfCollapsed & wbCollapseFlags to definitions lacking them.
[TES3] ALCH - Changed form type name to 'Alchemy Potion.'
[TES3] ALCH\TEXT - Changed name to 'Inventory Image Filename' to match the Construction Set.
[TES3] APPA - Reverted form type name to 'Alchemical Apparatus.'
[TES3] APPA,ARMO,BOOK,CLOT,INGR,LIGH,LOCK,MGEF,MISC,PICK, REPA, WEAP\ITEX - Changed name to 'Inventory Image Filename' to match the Construction Set.
[TES3] ARMO,CLOT,REFR,WEAP - Changed 'Enchanting Charge' to 'Enchantment Charge.'
[TES3] ARMO,CLOT\BNAM,CNAM - Simply named Male and Female respectfully.
[TES3] BOOK\BKDT - 'Is Scroll' simply changed to 'Scroll.'
[TES3] BSGN\TNAM - Names changed to 'Constellation Image Filename.'
[TES3] CELL\INTV,WHGT,DATA - Added SetAfterLoad to swap the old 'Water Height' signature to the new one and zero out interior cell's Data\Grid\X,Y.
[TES3] CELL\RGNN - Named changed to 'Region Weather Type'
[TES3] CELL\NAM5 - Named changed to 'Region Map Color'
[TES3] CREA\NPDT - Attack Sets have been restructed to be more readable.
[TES3] CREA,NPC_\AI Packages - Simplified the array and structs to be a bit more readable. Several name changes.
[TES3] FACT - Restructed and renamed multiple definitions to be more accurate and readable.
[TES3] GMST\STRV,INTV,FLTV - Simplifed names.
[TES3] INFO\DATA - Changed 'Gender' to 'Sex' to match the Construction Set.
[TES3] INFO\QSTN,QSTF,QSTR - Simplified names.
[TES3] LAND\VTEX - Fixed error that would cause out of order or unknown signatures when simplifed records was enabled. (VTXT -> VTEX)
[TES3] LAND\VTEX - Changed name to be more discriptive.
[TES3] NPC_\NPDT - Updated Union Decider funtion to work from the signature instead of the struct to simplify the defintion. Comment5*
[TES3] PGRD\DATA - Removed cpCritical as it doesn't apply.
[TES3] REGN\CNAM - Named changed to 'Region Map Color.'
[TES3] REGN\SNAM - Renamed to 'Region Sounds.'
[TES3] SKIL\SKDT - Replaced Actions array with a struct.
[TES3] SOUN\DATA - Added wbDiv to turn 'Volume' into a float to match the Construction Set.
[TES3] SPEL - Changed form type name to Spell to make it simple and align with other game definitions. For real this time.
[TES4] ALCH - Changed form type name to 'Alchemy Potion.'
[TES4] BSGN\ICON - Changed name to 'Constellation Image Filename.'
[TES4] CLAS\ICON - Changed name to 'Menu Image Filename.'

Kuroitsune commented:
Closing this for now. I can't finish what I was doing while recovering from hand surgery.
